services/dbaas/dbaas_client.go

```diff
@@ -19,102 +19,476 @@ package dbaas
+	c.xtradbClusterClient = controllerv1beta1.NewXtraDBClusterAPIClient(conn)
+	c.psmdbClusterClient = controllerv1beta1.NewPSMDBClusterAPIClient(conn)
+	c.logsClient = controllerv1beta1.NewLogsAPIClient(conn)
+
+	return nil
+}
+
 // CheckKubernetesClusterConnection checks connection with kubernetes cluster.
 func (c *Client) CheckKubernetesClusterConnection(ctx context.Context, kubeConfig string) (*controllerv1beta1.CheckKubernetesClusterConnectionResponse, error) {
-	return c.kubernetesClient.CheckKubernetesClusterConnection(ctx, &controllerv1beta1.CheckKubernetesClusterConnectionRequest{
-		KubeAuth: &controllerv1beta1.KubeAuth{Kubeconfig: kubeConfig},
-	})
+	responseCh := make(chan *apiResponse)
+	c.requests <- &apiRequest{
+		handler: func() (interface{}, error) {
+			in := &controllerv1beta1.CheckKubernetesClusterConnectionRequest{
+				KubeAuth: &controllerv1beta1.KubeAuth{
+					Kubeconfig: kubeConfig,
+				},
+			}
+			out, err := c.kubernetesClient.CheckKubernetesClusterConnection(ctx, in)
+			return out, err
+		},
+		responseCh: responseCh,
+	}
+	resp := <-responseCh
+	if resp.err != nil {
+		return nil, resp.err
+	}
+	return resp.out.(*controllerv1beta1.CheckKubernetesClusterConnectionResponse), resp.err
 }
 
 // ListXtraDBClusters returns a list of XtraDB clusters.
 func (c *Client) ListXtraDBClusters(ctx context.Context, in *controllerv1beta1.ListXtraDBClustersRequest, opts ...grpc.CallOption) (*controllerv1beta1.ListXtraDBClustersResponse, error) {
-	return c.xtradbClusterClient.ListXtraDBClusters(ctx, in, opts...)
+	responseCh := make(chan *apiResponse)
+	c.requests <- &apiRequest{
+		handler: func() (interface{}, error) {
+			return c.xtradbClusterClient.ListXtraDBClusters(ctx, in, opts...)
+		},
+		responseCh: responseCh,
+	}
+	resp := <-responseCh
+	if resp.err != nil {
+		return nil, resp.err
+	}
+	return resp.out.(*controllerv1beta1.ListXtraDBClustersResponse), resp.err
 }
 
 // CreateXtraDBCluster creates a new XtraDB cluster.
 func (c *Client) CreateXtraDBCluster(ctx context.Context, in *controllerv1beta1.CreateXtraDBClusterRequest, opts ...grpc.CallOption) (*controllerv1beta1.CreateXtraDBClusterResponse, error) {
-	return c.xtradbClusterClient.CreateXtraDBCluster(ctx, in, opts...)
+	responseCh := make(chan *apiResponse)
+	c.requests <- &apiRequest{
+		handler: func() (interface{}, error) {
+			return c.xtradbClusterClient.CreateXtraDBCluster(ctx, in, opts...)
+		},
+		responseCh: responseCh,
+	}
+	resp := <-responseCh
+	if resp.err != nil {
+		return nil, resp.err
+	}
+	return resp.out.(*controllerv1beta1.CreateXtraDBClusterResponse), resp.err
 }
 
 // UpdateXtraDBCluster updates existing XtraDB cluster.
 func (c *Client) UpdateXtraDBCluster(ctx context.Context, in *controllerv1beta1.UpdateXtraDBClusterRequest, opts ...grpc.CallOption) (*controllerv1beta1.UpdateXtraDBClusterResponse, error) {
-	return c.xtradbClusterClient.UpdateXtraDBCluster(ctx, in, opts...)
+	responseCh := make(chan *apiResponse)
```

Comment:
I think it's overcomplicated. Please check how API endpoints were disabled for Backup using interceptors. That should be enough.

Reply:
If required we can use mutexes.

Reply:
Done, services now have Enabled method.